lib/src/audio_file_waveforms.dart

```diff
@@ -88,233 +93,299 @@ class AudioFileWaveforms extends StatefulWidget {
 
 class _AudioFileWaveformsState extends State<AudioFileWaveforms>
     with SingleTickerProviderStateMixin {
-  late AnimationController animationController;
-  late Animation<double> animation;
-  double _animProgress = 0.0;
+  late AnimationController _growingWaveController;
+  late Animation<double> _growAnimation;
+
+  double _growAnimationProgress = 0.0;
   final ValueNotifier<int> _seekProgress = ValueNotifier(0);
   bool showSeekLine = false;
-  late List<int> _waveData;
+
   late EdgeInsets? margin;
   late EdgeInsets? padding;
   late BoxDecoration? decoration;
   late Color? backgroundColor;
   late Duration? animationDuration;
   late Curve? animationCurve;
-  late double? density;
   late Clip? clipBehavior;
   late PlayerWaveStyle? playerWaveStyle;
 
   @override
   void initState() {
     super.initState();
     _initialiseVariables();
-    _calculateWaveform().whenComplete(() {
-      animation.addListener(() {
-        if (mounted) {
-          setState(() {
-            _animProgress = animation.value;
-          });
-        }
-      });
-      animationController.forward();
-    });
-    animationController = AnimationController(
+    _growingWaveController = AnimationController(
       vsync: this,
       duration: widget.animationDuration,
     );
-    animation = Tween(begin: 0.0, end: 1.0).animate(CurvedAnimation(
-        parent: animationController, curve: widget.animationCurve));
-    PlatformStreams.instance.onDurationChanged.listen((event) {
-      if (widget.playerController.playerKey == event.playerKey) {
-        _seekProgress.value = event.type;
-        _updatePlayerPercent(widget.size);
+    _growAnimation = Tween(begin: 0.0, end: 1.0).animate(CurvedAnimation(
+      parent: _growingWaveController,
+      curve: widget.animationCurve,
+    ));
+
+    _growingWaveController.forward();
+    _growAnimation.addListener(() {
+      if (mounted) {
+        setState(() {
+          _growAnimationProgress = _growAnimation.value;
+        });
       }
     });
+    widget.playerController.onCurrentDurationChanged.listen((event) {
+      _seekProgress.value = event;
+      _updatePlayerPercent(widget.size);
+    });
+
+    if (widget.waveformData.isNotEmpty) {
+      _addWaveformData(widget.waveformData);
+    } else {
+      if (widget.playerController.waveformData.isNotEmpty) {
+        _addWaveformData(widget.playerController.waveformData);
+      }
+      if (!widget.continuousWaveform) {
+        widget.playerController.addListener(() {
+          _addWaveformData(widget.playerController.waveformData);
+        });
+      } else {
+        widget.playerController.onCurrentExtractedWaveformData
+            .listen(_addWaveformData);
+      }
+    }
   }
 
   @override
   void dispose() {
-    animation.removeListener(() {});
-    animationController.dispose();
+    widget.playerController.removeListener(() {});
```

Comment:
This doesn't make any sense. Use same function as line no 145 as argument of removeListener.